src/features/tokenGatedRooms/sagas/getTokenMetadata.saga.ts

```diff
@@ -1,31 +1,119 @@
+        symbol: await instance.symbol(),
+        uris: await Promise.all(
+            tokenIds.map((tokenId) => instance.tokenURI(BigNumber.from(tokenId)))
+        ),
+    }
+}
+
+async function getERC777TokenMetadata(tokenAddress: Address, provider: Provider) {
+    const instance = getERC777(tokenAddress, provider)
+
+    return {
+        name: await instance.name(),
+        symbol: await instance.symbol(),
+        granularity: BigNumber.from(await instance.granularity()),
+    }
+}
+
+async function getERC1155TokenMetadata(
+    tokenAddress: Address,
+    provider: Provider,
+    tokenIds: string[]
+) {
+    const instance = getERC1155(tokenAddress, provider)
+
+    return {
+        uris: await Promise.all(
+            tokenIds.map((tokenId) => instance.tokenURI(BigNumber.from(tokenId)))
+        ),
+    }
+}
+
 function* onGetTokenMetadata({
-    payload: { tokenAddress, tokenType, provider },
+    payload: { tokenAddress, tokenIds, tokenType, provider },
 }: ReturnType<typeof TokenGatedRoomAction.getTokenMetadata>) {
     try {
-        if (tokenType.standard !== TokenTypes.ERC20.standard) {
-            throw new Error(`Unsupported token type: ${tokenType.standard}`)
-        }
-        const instance = getERC20(tokenAddress, provider)
+        let metadata: TokenMetadata
 
-        const name: string = yield instance.name()
-
-        const symbol: string = yield instance.symbol()
-
-        const decimals: BigNumber = yield instance.decimals()
+        switch (tokenType.standard) {
+            case TokenTypes.ERC20.standard:
+                metadata = yield getERC20TokenMetadata(tokenAddress, provider)
+                break
+            case TokenTypes.ERC721.standard:
+                if (tokenIds.length === 0) {
+                    throw new Error('Token ID is required for ERC721 tokens')
+                }
+                metadata = yield getERC721TokenMetadata(tokenAddress, provider, tokenIds)
+                break
+            case TokenTypes.ERC777.standard:
+                metadata = yield getERC777TokenMetadata(tokenAddress, provider)
+                break
+            case TokenTypes.ERC1155.standard:
+                if (tokenIds.length === 0) {
+                    throw new Error('Token ID is required for ERC1155 tokens')
+                }
+                metadata = yield getERC1155TokenMetadata(tokenAddress, provider, tokenIds)
+                break
+            default:
+                throw new Error(`Unsupported token type: ${tokenType.standard}`)
+        }
 
         yield put(
-            TokenGatedRoomAction.setERC20Metadata({
-                name,
-                symbol,
-                decimals,
+            TokenGatedRoomAction.setTokenMetadata({
+                name: metadata.name,
+                symbol: metadata.symbol,
+                decimals: metadata.decimals?.toString(),
+                granularity: metadata.granularity?.toString(),
+                uri: metadata.uri,
```

Comment:
@bearni95 I'm making changes in this file. Please confirm: `uri` here should be `uris` (1 per token id)?

Reply:
Yes! Probably what makes more sense is to keep it as a key-value, `tokenId` --> `uri`